contrib/inventory/azure_rm.py

```diff
@@ -300,24 +307,47 @@ def __init__(self, args):
         self.log("setting subscription_id")
         self.subscription_id = self.credentials['subscription_id']
 
+        # get authentication authority
+        # for adfs, user could pass in authority or not.
+        # for others, use default authority from cloud environment
+        if self.credentials.get('authority') is None:
+            self._authority = self._cloud_environment.endpoints.active_directory
+        else:
+            self._authority = self.credentials.get('authority')
+
+        # get resource from cloud environment
+        self._resource = self._cloud_environment.endpoints.active_directory_resource_id
+
         if self.credentials.get('client_id') is not None and \
            self.credentials.get('secret') is not None and \
            self.credentials.get('tenant') is not None:
             self.azure_credentials = ServicePrincipalCredentials(client_id=self.credentials['client_id'],
                                                                  secret=self.credentials['secret'],
                                                                  tenant=self.credentials['tenant'],
                                                                  cloud_environment=self._cloud_environment)
-        elif self.credentials.get('ad_user') is not None and self.credentials.get('password') is not None:
-            tenant = self.credentials.get('tenant')
-            if not tenant:
-                tenant = 'common'
-            self.azure_credentials = UserPassCredentials(self.credentials['ad_user'],
-                                                         self.credentials['password'],
-                                                         tenant=tenant,
-                                                         cloud_environment=self._cloud_environment)
+
+        elif self.credentials.get('ad_user') is not None and \
+                self.credentials.get('password') is not None:
+                tenant = self.credentials.get('tenant')
+                if not tenant:
+                    tenant = 'common'  # SDK default
+
+                client_id = self.credentials.get('client_id')
+                if not client_id:
+                    client_id = '04b07795-8ddb-461a-bbee-02f9e1bf7b46'
```

Comment:
@johanste is this kosher? Seems strange for us to embed the Azure CLI's client_id here...

Reply:
@nitzmahone @johanste thanks for the comments. I'd like to share more on using the client_id of az cli.
1. before the change, ansible is using UserPassCredential from azure.common.credentials. azure.common.credentials uses az cli client_id, [code here](https://github.com/Azure/msrestazure-for-python/blob/v0.4.23/msrestazure/azure_active_directory.py#L355-L359)

1. UserPassCredential is obsolete, and not working for adfs scenario, then we switch to adal. adal requires client_id.
1. we tried to register ansible as multiple-tenant in adal to get own client id, but didn't find way to do native app register without consent.
1. if not using default client id, it will cause backward compatibility issue for existing username/password users.

pls advice how to register ansible as a multi-tenant app in adal without owner consent during authentication, so we can have our own client id.